### PR TITLE
NL: Improve fallbacks

### PR DESCRIPTION
**Pull request by pradh**
1. Switch to having a default place, but report that we are using a default place
2. Propagate specific message for comparison fallback, which has multiple places
3. Generally simplify comparison flow to *only* use the most recent place(s)
4. Disable fallback for default place

Wrote up the [fallback matrix here](https://docs.google.com/document/d/13-1zsYZm_RFkzhKb4GxVy7DQNU8ZiEYZSY8001R-GsU/edit).

Some screenshots:

[agriculture workers]
image

[GDP of China and USA] -> [compare with India]
image

[GDP of China and India] -> [compare fertility rate instead]
image

### CODE REVIEW

File: server/lib/nl/common/utterance.py [hunk 1]
```diff
@@ -93,10 +93,22 @@ class ChartType(IntEnum):
 
 class FulfillmentResult(str, Enum):
   """Where is a variable or place from?"""
+  # From the current user provided query.
   CURRENT_QUERY = "CURRENT_QUERY"
+  # From a past user query in the context.
+  # Importantly, no place/variable was recognized in the current query.
```
Comment (jehangiramjad) [inline server/lib/nl/common/utterance.py:99]: What does this comment mean? Is ``PAST_QUERY`` only set if no place/variable was recognized int he current query?

**Review by jehangiramjad — APPROVED** [review]
Thanks!

File: server/lib/nl/common/utterance.py (hunk 1 above)
Comment (pradh) [reply server/lib/nl/common/utterance.py:99]: Correct.

**Review by pradh — COMMENTED** [review]
Thanks Jehangir!

**Pull request merged**